### ISSUE TITLE
[HOLD for payment 2024-08-15] [$250] Can't enter maximum tax amount

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#42601]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.75-0
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @DylanDylann 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1716540690746529

## Action Performed:
Pre requisite: Have a control workspace with tax enabled and with a rate of 12%
1. As a employee click "submit expense" in the workspace
2. Enter amount đ20000 (currency: VND)
3. Select tax rate: 12%
4. Edit tax amount to đ2143
## Expected Result:
The error message : "Maximum tax amount is ₫2,143"  should appear only when we enter a mount greater than "đ2143"

## Actual Result:
The error message : "Maximum tax amount is ₫2,143" appear when we enter exactly "đ2143"

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/9a49c05c-4701-4bf0-b667-8d8b3bd6ae45

https://github.com/Expensify/App/assets/38435837/c7a3dd92-726f-4096-85f6-985d4cb6a1b9

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0119a2144ed665fa18</li>
        <li>Upwork Job ID: 1800124610279105511</li>
        <li>Last Price Increase: 2024-06-10</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Reviewer | 102728051</li>
<li>ShridharGoel | Contributor | 102728052</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

**Comment by DylanDylann** [comment]
I reported this bug from another PR. Could I take over this issue as C+?

**Comment by muttmuure** [comment]
Catching up from OOO

**Comment by muttmuure** [comment]
Working through higher priorities right now

**Comment by DylanDylann** [comment]
@muttmuure Should this issue be handled externally?

**Comment by muttmuure** [comment]
Thanks for your patience, looking at this now

**Comment by muttmuure** [comment]
OK cool, that seems like something we should fix. If we can handle it externally, that is my preference

**Comment by DylanDylann** [comment]
I need to clarify how we calculate the tax amount. Asked in [Slack thread](https://expensify.slack.com/archives/C02NK2DQWUX/p1718183483805599)

**Comment by muttmuure** [comment]
Not overdue

**Comment by DylanDylann** [comment]
@ShridharGoel [Your proposal](https://github.com/Expensify/App/issues/42601#issuecomment-2132176882) looks good to me. We should render the tax amount based on the decimal of each currency

**Comment by DylanDylann** [comment]
@ShridharGoel [Your proposal](https://github.com/Expensify/App/issues/42601#issuecomment-2132176882) looks good to me. We should render the tax amount based on the decimal of each currency

🎀👀🎀 C+ reviewed

**Comment by Tony-MK** [comment]
@DylanDylann , could you kindly explain to me why issue occurs with other currencies like USD.

Thanks

**Comment by Tony-MK** [comment]
@DylanDylann , could you kindly explain to me why issue occurs with other currencies like USD as well.

Thanks

**Comment by Tony-MK** [comment]
@DylanDylann , could you kindly explain to me why issue occurs with other currencies like USD as well.

Thanks

**Comment by francoisl** [comment]
> To be discussed: 12% of 20000 is 2142.86 so we can discuss if we want to use 2143 as the max tax or do we want to lower it to 2142. I think 2143 should be used.

Rounding up makes sense to me here.

> Also, the tax amount is set to 2142.86 by default in the tax form while editing, but it should have been 2143.

Yeah let's just make sure to be consistent, at the moment it appears that we don't round the amount the same way in the expense view and the tax amount view:

https://github.com/Expensify/App/assets/2229301/72147edf-374a-4712-97da-9f0556dd6b48

**Comment by DylanDylann** [comment]
@ShridharGoel How is it going?

**Comment by muttmuure** [comment]
Waiting on linked pr

**Comment by DylanDylann** [comment]
@ShridharGoel Any update?

**Comment by DylanDylann** [comment]
@ShridharGoel Could you give ETA for the PR?

**Comment by DylanDylann** [comment]
The PR was deployed 4 days ago

**Comment by francoisl** [comment]
cc @muttmuure ^

**Comment by mountiny** [comment]
@anmurali Matt is ooo this week and this failed with automation after a deploy, this is ready for payment

$250 to @ShridharGoel and to @DylanDylann

**Comment by anmurali** [comment]
@DylanDylann is paid
@ShridharGoel - your offer is [here](https://www.upwork.com/nx/wm/offer/103539188)

**Comment by muttmuure** [comment]
Waiting for @ShridharGoel to accept the offer

**Comment by francoisl** [comment]
@ShridharGoel can you accept the offer so we can process the payments and close this one please?

**Comment by DylanDylann** [comment]
Not overdue. Waiting for @ShridharGoel

**Comment by anmurali** [comment]
Paid.
